runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -527,7 +527,7 @@ def up(self, verbose: bool = True, force: bool = False):
             logger.info("Launching cluster with Den")
             DenLauncher.up(cluster=self, verbose=verbose, force=force)
 
-        else:
+        elif self.launcher_type == LauncherType.LOCAL:
```

Comment:
L109 in this file should be `self.launcher_type = launcher_type or configs.launcher_type`